storage/types.go

```diff
@@ -156,16 +168,19 @@ func (n *NodeID) SetBit(i int, b uint) {
 
 // Bit returns 1 if the ith bit is true, and false otherwise.
 func (n *NodeID) Bit(i int) uint {
-	bIndex := (n.PathLenBits - i - 1) / 8
+	if got, want := i, n.PathLenBits()-1; got > want {
+		panic(fmt.Sprintf("storage: Bit(%v) > (PathLenBits -1): %v", got, want))
```

Comment:
I'm not overjoyed about all these panics in low level code. It's duplicating the bounds check on the array with a chance to get it wrong.